### ISSUE TITLE
[NEW FEATURE] New threading mode - no threads

### ISSUE THREAD

**Issue by silvioprog** [risoflora/libsagui#9]
Since the fourth threading mode of the core allows to disable the internal threads, it would be nice to provide it in the Sagui library too. It would be useful for those applications that already declares a main loop, so the programmers could use it to periodically process the requests and disable the internal library thread/loop.

**Comment by silvioprog** [comment]
Study progress:

1. it is possible to implement the feature by allowing an external loop which could be managed by `select()`/`epoll()`.
2. since `select()`/`epoll()` is not cross platform in some compilers/languages, we could provide a function `sg_process()` that could be called within external loops (application).

Problems, obstacles ...:

3. it is quite difficult to implement the features without breaking the current ABI, and it's a little bit early to release another major version.
4. by design, maybe the feature couldn't be implemented in a single function, then it would grows the final binary anyway.
5. I've tried to use the MHD's suspend/resume, however I've got an infinity lock in the client socket when tried to combine it with pthread-like threads and `select()` (asked [here](http://lists.gnu.org/archive/html/libmicrohttpd/2019-03/msg00004.html) plus small spelling correction [here](http://lists.gnu.org/archive/html/libmicrohttpd/2019-03/msg00005.html)).

In short, this feature depends on 5., so, if that is solved, the feature will be implemented.

Another idea was issued [here](https://bugs.gnunet.org/view.php?id=5653). It would be very useful to reuse the internal library loop.

**Comment by silvioprog** [comment]
This feature is unviable to be implemented due to possible blocking calls! An initial implementation was done is [this commit](https://github.com/risoflora/libsagui/commit/041b2601f36cacfb0b4bf4690c93096c36190d34), but it locks the entire server when the client calls a route calling a blocking routine. Since MHD already handles it automatically, we are going to let it as it already is.
